ADR/0012-namespace-name-format.md

```diff
@@ -0,0 +1,23 @@
+# 12. Record architecture decisions
+
+Date: 2023-01-23
+
+## Status
+
+Accepted
+
+## Context
+
+The OSD-based control plane provisions one namespace in the target member cluster for every workspace (internally represented by a Space CR) which is created for a StoneSoup user. All the namespace names provisioned in this way should have a fixed suffix because of two reasons:
+1. Visual separation of the namespaces provisioned for StoneSoup workspaces.
+2. Limiting the risk of conflicting with the names used for other namespaces that are present in the cluster - either by default for every OCP/OSD/ROSA cluster or created via other SRE pipelines.
+
+## Decision
+
+Every namespace provisioned for a StoneSoup top-level workspace will have a name with the fixed suffix `-tenant`. The complete format will be `<workspace-name>-tenant`.
```

Comment:
Where does the "workspace-name" come from?

Reply:
Currently the workspace name matches the "complaint" username of the user who signed up for Stonesoup. In most cases the "complaint" username is just the soo username. But in case of usenames which are not DNS-1123 complaint then the original sso name is modified. For example an sso username `john@domain.com` will be transformed to `john` complaint username, so the corresponding Workspace/Space name will be `john` and the corresponding workspace/space namespace will be `john-tenant`.

Reply:
Yeah, the process of getting the workspace name from the username is pretty complex (for the default workspaces created automatically by our operators). I described some parts of it in the slack thread: https://redhat-internal.slack.com/archives/C02CTEB3MMF/p1674489259588719?thread_ts=1674486159.298199&cid=C02CTEB3MMF
I didn't see it as important for the context of this document (as it's our operator's implementation detail). For this doc it's just a workspace name - doesn't matter if it's based on username, [SpaceRequest API](https://docs.google.com/document/d/1uqgghk1lN9dyoBLsvn5YD443TwKrsKHU6fvfUuo84Hs/edit#heading=h.mufidl7h27i0), or manually provided by a user.